packages/std/src/math/decimal.rs

```diff
@@ -82,35 +81,35 @@ impl Decimal {
     /// Returns `None` if the internal multiplication overflows.
     fn sqrt_with_precision(&self, precision: u32) -> Option<Self> {
         let inner_mul = 100u128.pow(precision);
-        self.0.checked_mul(inner_mul).map(|inner| {
+        self.0.checked_mul(inner_mul.into()).ok().map(|inner| {
             let outer_mul = 10u128.pow(9 - precision);
-            Decimal(inner.isqrt() * outer_mul)
+            Decimal(inner.isqrt().checked_mul(Uint128::from(outer_mul)).unwrap())
         })
     }
 }
 
-impl Fraction<u128> for Decimal {
+impl Fraction<Uint128> for Decimal {
```

Comment:
Can we implement both `Fraction<u128>` and `Fraction<Uint128>` to avoid the break?

We can remove `impl Fraction<u128>` in 1.0

Reply:
Hm. Having both of those makes things a little less handy.

```
error[E0283]: type annotations needed
   --> packages/std/src/math/decimal.rs:378:22
    |
378 |             fraction.numerator(),
    |             ---------^^^^^^^^^--
    |             |        |
    |             |        cannot infer type for type parameter `T` declared on the trait `Fraction`
    |             this method call resolves to `T`
```

And then we have to...
```diff
- fraction.numerator(),
+ <_ as Fraction<Uint128>>::numerator(&fraction),
```

Reply:
If we make `Decimal` generic, we can probably turn the `Fraction` type parameter into an associated type. But then we're looking at more breakage :D

Reply:
How about we stick with `Fraction<u128>` only for now? I have another branch waiting with a generic `Decimal` thing - we can see how that will affect things and if we want it.

Reply:
> How about we stick with `Fraction<u128>` only for now?

Sounds solid. We can change between 0.16 and 1.0 of we prefer `Fraction<Uint128>`.